From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 24 Jan 2024 23:32:37 +0100
Subject: [PATCH] Read packs - Content - Definitions - Read before bootstrap

Invokes the reading of all content definitions of all types from all packs before the server bootstrap is called.
This relies on the `FiddleContentDefinitionReader` defined in the "Read packs - Content - Definitions - Implementation to read all" patch.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 89f95494d276f54fe753eb3caa35e2cc58c1ffa3..9f2c74ca4fbb740e298c88ae47a1a36a8436582f 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -152,6 +152,16 @@ public class Main {
             }
             // Fiddle end - read packs - prevent duplicate packs
 
+            // Fiddle start - read packs - content definitions - read before bootstrap
+            org.fiddlemc.fiddle.pack.read.fiddlecontent.definition.FiddleContentDefinitionsByType readContentDefinitions;
+            try {
+                readContentDefinitions = org.fiddlemc.fiddle.pack.read.fiddlecontent.definition.FiddleContentDefinitionReader.readAndSortDefinitions();
+            } catch (Exception e) {
+                LOGGER.error("Could not read block and item definitions", e);
+                return;
+            }
+            // Fiddle end - read packs - content definitions - read before bootstrap
+
             io.papermc.paper.plugin.PluginInitializerManager.load(optionset); // Paper
             Bootstrap.bootStrap();
             Bootstrap.validate();
